src/Stream.php

```diff
@@ -318,14 +318,18 @@ public function getMetadata(?string $key = null)
      */
     private function setStream($stream, string $mode = 'r'): void
     {
+        $error    = null;
         $resource = $stream;
 
         if (is_string($stream)) {
             try {
                 $resource = fopen($stream, $mode);
             } catch (Throwable $error) {
+            }
+
+            if (! is_resource($resource)) {
```

Comment:
This is already handled by the next check with `isValidStreamResourceType`. This change should not be necessary.

Reply:
Different exception types are expected depending on the failure. Runtime exception is expected by interop integration tests when the file path does not exist for example

Reply:
Then I think it should be two separate exceptions. `Invalid stream reference provided: ` does not make much sense. 

That try-catch block is pretty much only for the empty string throwing ValueError and may be for the mezzio error handler converting warnings to exceptions.
What about the warning emitted by fopen on non-existent path? There was error handler but it was removed for whatever reason. Does interop say something about that?

Reply:
The error handler was removed in #103 - I'm unsure exactly why, but the effect is that `fopen` warnings are no longer silenced.

The try/catch does indeed only catch the empty string case.

Considering that the exception is thown for both empty string and non-existent path, the contract with Interop for throwing `\RuntimeException` is satisfied, and I think it's preferable to expose the PHP warnings to users rather than swallow them in the exception.

What do you suggest the change should be here? Re-introduce the error handler? Change the exception message?

Reply:
It seems removing the warning to exception conversion was an unrelated change. Adding it back here would be an unrelated change again. 

I think exception message should be improved to include the offending string and it would be sufficient for now.